### DISCUSSION TITLE
About SpikeTime training of NMNIST

### DISCUSSION THREAD

**Discussion by ahmetakman** [lava-nc/lava-dl#51] [General]
Hello,
I am curious about something. The tutorial on the repo uses SpikeRate for the NMNIST training. And it was said " Since the target spike train is not known for this problem, we use `SpikeRate` loss and target high spiking rate for true class and low spiking rate for false class. "
So, if we assign some arbitrary distinct 10 spike trains as target spike trains, would not it be better to use the SpikeTime method which may provide better efficiency and latency. Am I wrong with my assumption? 

Sincerely,

**Comment by bamsumit** [comment]
If you use `SpikeTime`, you are asking the network to match the 10 spike times exactly whereas if you are using `SpikeRate` corresponding to 10 spikes per sample interval, then 10 spikes anywhere is a valid solution. The `SpikeTime` solution is a special case of `SpikeRate` solution. This means that using rate, the final solution is flexible and, therefore, easier to train.

**Comment by ahmetakman** [comment]
Thank you very much for the explanation.
